File: payment-service/src/main/java/com/example/payment_service/model/event/RideCompletedEvent.java
package com.example.payment_service.model.event;

import lombok.Builder;

import java.math.BigDecimal;
import java.util.UUID;

@Builder
public record RideCompletedEvent(
        UUID rideId,
        UUID driverId,
        UUID passengerId,
        BigDecimal amount,
        UUID paymentMethodId,
        String currency
) {
}

File: driver-service/src/main/kotlin/com/example/driver_service/model/event/RideCompletedEvent.kt
package com.example.driver_service.model.event

import java.math.BigDecimal
import java.util.UUID

data class RideCompletedEvent(
    val rideId: UUID,
    val driverId: UUID,
    val passengerId: UUID,
    val amount: BigDecimal,
    val paymentMethodId: UUID,
    val currency: String
)
